jupyterlab_gcpscheduler/src/components/scheduler_form.tsx

```diff
@@ -181,14 +181,16 @@ export class InnerSchedulerForm extends React.Component<
   private prepopulateImageUri() {
     this.props.gcpService.getImageUri().then((retrievedImageUri: string) => {
       if (retrievedImageUri || this.props.values.imageUri) {
-        const imageUri = retrievedImageUri
+        let imageUri = retrievedImageUri
           ? retrievedImageUri
           : this.props.values.imageUri;
-        const lastColonIndex = imageUri.lastIndexOf(':');
-        const matchImageUri =
-          lastColonIndex !== -1 ? imageUri.substr(0, lastColonIndex) : imageUri;
-        const matched = ENVIRONMENT_IMAGES.find(i => {
-          return String(i.value).startsWith(matchImageUri);
+        if (imageUri === CUSTOM_CONTAINER.value) {
+          imageUri = this.props.values.customContainerImageUri;
+        }
+        const matched = ENVIRONMENT_IMAGES.slice(1).find(env => {
```

Comment:
Correct me if I am mistaken, but it looks like the algorithm here is to loop through each environment image until you find one where all keywords are present in the `imageUri`? If that's the case, it might be slightly more readable to use a traditional `for ... of` loop through `ENVIRONMENT_IMAGES`. Then, you can use something like `env.searchKeywords.every(k => String(imageUri).includes(k))` and if true you can break the loop.

It's probably just me but the negation of the `env.searchKeywords.find` and the include inside mixed me up a bit.

Reply:
Fixed.